### PR TITLE
Use mozjpeg instead of jpegtran

### PR DESCRIPTION
**Pull request by pavelloz**
Hi,

I experimented with mozjpeg and it looks like it produces much better results than jpegtran.

jpegtran:
```
    f72836bbac59ca5e265a0fc7591dfd09.jpg  2.43 MB          [emitted]
```
Mozjpeg:
```
    b47f5167a0c34fca03f17dc35df9358d.jpg  1.21 MB          [emitted]
```

I couldn't see much of a difference on 300% zoom when i was testing on quality 75, but i set the default quality to 80 anyway. Hope you like it, I sure do like those smaller images.

PS. Dont mind the whitespace diff, it's just something my editor does with trailing spaces.

### CODE REVIEW

**Comment by NicholasTuck** [comment]
+1 to the idea.
I attended a conference talk @ Velocity Conf this year by Tobias Baldauf from Akamai where he breaks down the differences and highlights mozjpeg quite a bit.  Including progressive jpegs and some ways he has optimized it further. Pretty cool stuff.

I would expect the progressive options still be available. I didn't dig into your code but saw that config removed.

**Comment by pavelloz** [comment]
Yes, progressive option available and it is set to `true` by default which is pretty useful for optimizing perceived performance on web.

Source: https://github.com/imagemin/imagemin-mozjpeg#api

**Comment by pavelloz** [comment]
After watching the [talk](https://www.youtube.com/watch?v=vCHkpxPP6Es) metioned by @NicholasTuck I decided that leaving default quality (75) is more optimal choice than bumping it up to 80. It looks like [dssim (dissimilarity) index](https://github.com/pornel/dssim) is very low even on 75 setting.

So the actual result is now:
```
    04c36c33fe0c2dfec1e4db6393bc9f81.jpg  1.04 MB          [emitted]
```

And the dssim index for this particular test image is:
```
0.00347507	output.png
```

And according to the study mentioned in the talk, 0.015 is still acceptable by end users. 
Because there is only one test image in the repo I don't want to tinker down it. To be sure about actual we would need to have more input data to compare.

To sum up, at the moment, mean from the study made by Tobias Baldauf is that quality 75 provided around 0.09% dssim in 2014, so we are good to go with the default setting.

**Comment by NicholasTuck** [comment]
Glad I could help.  I should have linked to the video, I didn't know they would be out yet to be honest. I found his most recent one I went to here: https://www.youtube.com/watch?v=66JINbkBYqw (June 2016). I haven't watched the 2014 yet.
The last part is about http2 which is interesting though not relevant yet. But the first half covers much of what you mentioned in the other one.

**Comment by tcoopman** [comment]
@pavelloz I don't have much time at the moment. Your PR looks good. If you want to I can give you access to this repo so you can merge the PR yourself, and if you have the time, you can take a look at the other PRs too?

**Comment by pavelloz** [comment]
@tcoopman I don't know also if i will be able to contribute consistently, I just thought that was very important feature have.

Of course I can try to provide as much value as I can with my limited resources.

Just want to be fair and communicate it up front to avoid disappointment :)

**Comment by tcoopman** [comment]
@pavelloz I gave you access, so feel free to merge. Let me know if you want to create a new release on npm too.

As for your time, just see what's possible for you. I'm also going to create an issue to see if anyone else is interested in helping maintain this package.

**Comment by tcoopman** [comment]
@pavelloz regarding this PR, it would be nice if you could start a changelog file. To say that you've added this.

**Comment by pavelloz** [comment]
Sure :)

I'm just learning about npm packages publishing, never done that before so I guess its about time.

I get from the tags that I should use SemVer?

**Comment by tcoopman** [comment]
@pavelloz yeah, this is a breaking change in the config if I read it correctly, so we should bump to 3.0.0.

**Comment by pavelloz** [comment]
After some refinements and updates to meta files I'm merging this PR.

Im pretty new to the packages and webpack loaders development so please correct me if i broke something unintentionally :)

I marked this version as 3.0.0 because of breaking changes in the API.

**Comment by pavelloz** [comment]
@tcoopman its probably a good time to release new version to npm :)

**Comment by tcoopman** [comment]
@pavelloz done. Thanks.
